### PR TITLE
Implement redesign of `<Hero />` 

### PR DESCRIPTION
**Pull request by nandereck**
🎟️ [Asana Task](https://app.asana.com/0/1201874072258661/1203137429559227/f)
🔍 [Preview Link](https://react-components-git-nahero-redesign-hashicorp.vercel.app/components/hero)

---

## Description
This PR updates the `<Hero />` to match the updated [WPL spec](https://www.figma.com/file/VvpEQaWhKQExx9QTWRyayd/Patterns?node-id=2591%3A14651&t=fQdpqIOCikoalpLj-1). This is both a big departure from the current implementation and not backwards compatible (thus the `major` bump in the changeset). Given this and its [usage on dot com](https://hashicorp-mktg.sourcegraph.com/search?q=context%3Aglobal+%40hashicorp%2Freact-hero&patternType=standard&groupBy=repo), the current implementation will be copied into dot com (likely as `<LegacyHero />`). 

While this isn't ideal, the component is used on numerous pages in ways that are not compatible with the newest version. Updating those instances would require a fair amount of coordination with various stakeholders and design to gracefully transition every page (gathering updated content / assets, ensuring the new component meets requirements, etc.).

## Validation Steps
- Navigate to [the preview](https://react-components-git-nahero-redesign-hashicorp.vercel.app/components/hero)
- In dev tools, delete the `sidebar` element and make the `stage` element span the full width of the page.
- Verify that the component matches the [spec](https://www.figma.com/file/VvpEQaWhKQExx9QTWRyayd/Patterns?node-id=2591%3A14651&t=fQdpqIOCikoalpLj-1).

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

**Comment by nandereck** [comment]
@alexcarpenter - I merged the Solutions implementation with the implementation in https://github.com/hashicorp/react-components/pull/827. Mind giving this a look-over?

**Review by alexcarpenter — COMMENTED** [review]
Can we clean up some of the dependancies listed in the hero package.json?

```
"dependencies": {
    "@hashicorp/js-utils": "^1.0.10",
    "@hashicorp/platform-product-meta": "^0.1.0",
    "@hashicorp/react-actions": "^0.4.2",
    "@hashicorp/react-alert": "^6.0.2",
    "@hashicorp/react-button": "^6.0.4",
    "@hashicorp/react-image": "^4.0.4",
    "@hashicorp/react-text-input": "^5.0.1",
    "classnames": "^2.3.1",
    "formik": "^2.2.9",
    "promise-polyfill": "^8.2.1",
    "query-string": "^6.14.1"
  },
```

**Comment by nandereck** [comment]
> Can we clean up some of the dependancies listed in the hero package.json?

Oops, [done](https://github.com/hashicorp/react-components/pull/833/commits/2d289963a5427eb3b9ff588d69c49f0457c8b7a6)!

File: packages/hero/index.tsx [hunk 1]
```diff
@@ -39,7 +39,7 @@ const Hero = ({
               <InlineVideo
                 solution={solution}
                 url={url}
-                gradientPosition={false}
```
Comment (alexcarpenter) [inline packages/hero/index.tsx]: It should be in the latest release https://github.com/hashicorp/react-components/blob/main/packages/inline-video/types.ts#L6

Comment (nandereck) [reply packages/hero/index.tsx]: Yup - hadn't updated the package. Should be good here now!

File: packages/hero/types.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { ActionsProps } from '@hashicorp/react-actions/types'
+import { EyebrowProps } from './eyebrow-with-pattern'
+import { InlineVideoProps } from '@hashicorp/react-inline-video/types'
+import { Products } from '@hashicorp/platform-product-meta'
+
+type WithOptional<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>
+
+export type ContentProps = WithOptional<InlineVideoProps, 'url'> &
```
Comment (alexcarpenter) [inline packages/hero/types.ts:8]: Non-blocking thought, but can we make the prop name for the video be `videoUrl` vs `url`?

**Review by alexcarpenter — APPROVED** [review]

**Pull request closed without merging**